### DISCUSSION TITLE
GitHub Pages site no longer deploying on update

### DISCUSSION THREAD

**Discussion by jaredkrinke** [community/community#135679] [Other Feature Feedback, Questions, & Ideas]
**Is there any reason why a working GitHub Pages setup might have stopped working sometime between June 30 and August 12?** I don't recall any notifications of changes (edit: wrong!).

# Background
I have raw HTML being deployed from a selected branch (no Actions or build step--"Classic Pages experience") and everything has been working for a few years (nearly 200 deployments), but when I went to update the site yesterday, no deployment got triggered. I verified the commit was made to the correct branch, and even made a second commit, but neither update has triggered a deployment.

# Settings
On my Pages settings, there is a confusing note that mentions Jekyll--it's confusing because I *don't* use or need Jekyll:

> Actions is currently unavailable for your repository, and your Pages site requires a Jekyll build step. To continue building your site on pushes, you need to enable Actions. Learn how to enable GitHub Actions.

I'm not sure, but I think this message was always there (edit: wrong!), so I don't think it's relevant (edit: wrong!), but I could be wrong.

Any troubleshooting suggestions welcome!

**Comment by ansh3108** [comment]
Same thing happened with me, re-deploying it or using vercel worked for me. Vercel is pretty reliable and fast!

**Comment by jaredkrinke** [comment]
Turns out that the error message I pasted in was actually relevant.

**All I had to do was enable GitHub Actions with default settings, update one last time, and that triggered a successful deployment.** So I guess enabling GitHub Actions became a requirement for using Pages sometime in the last two months.

Edit: And it turns out there *was* an email communication about this, titled "Pages Legacy Worker Sunset"; excerpt:

>  we will be phasing out our legacy pagesworker architecture by the end of June 2024.
> ...
> Going forward, GitHub Actions will become a requirement for deploying Pages sites **that require a build step**
> ...
> If your Pages site is being built by the legacy infrastructure, a warning message will be displayed in your Pages settings starting today.

The confusing part is that my site *does not* require a build step, so I had interpreted this mail to mean that I would be unaffected. But apparently static HTML counts as a Jekyll build step. Seems odd, but I'm not planning to investigate any further.

**Comment by Quuxplusone** [comment]
> The confusing part is that my site does not require a build step, so I had interpreted this mail to mean that I would be unaffected. But apparently static HTML counts as a Jekyll build step. Seems odd, but I'm not planning to investigate any further.

I just ran into this too, for a gh-pages branch that was just a bunch of static files. I finally tracked down [this GitHub documentation](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#static-site-generators):

> If you publish your site from a source branch, GitHub Pages will use Jekyll to build your site by default. If you want to use a static site generator other than Jekyll, we recommend that you write a GitHub Actions to build and publish your site instead. Otherwise, disable the Jekyll build process by creating an empty file called `.nojekyll` in the root of your publishing source, then follow your static site generator's instructions to build your site locally.

However, it turns out that even with a `.nojekyll` file, you _still_ need to enable GitHub Actions — even though your "build step" is a no-op! See [this documentation](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#troubleshooting-publishing-from-a-branch):

> Your GitHub Pages site will always be deployed with a GitHub Actions workflow run, even if you've configured your GitHub Pages site to be built using a different CI tool. Most external CI workflows "deploy" to GitHub Pages by committing the build output to the gh-pages branch of the repository, and typically include a .nojekyll file. When this happens, the GitHub Actions workflow will detect the state that the branch does not need a build step, and will execute only the steps necessary to deploy the site to GitHub Pages servers.

I have verified that even with `.nojekyll`, the gh-pages branch fails to deploy until you enable GitHub Actions.

My conclusion is that you need to enable GitHub Actions even to deploy a plain HTML website these days. But if deployments are rare, then you might want to leave GitHub Actions disabled most of the time — just enable it manually right before you push and disable it a few minutes afterward.

Omitting `.nojekyll` means that GitHub will pay to run Jekyll on your static site, which will be a no-op. Adding `.nojekyll` will skip that no-op Jekyll step (in my test just now it decreased my deployment step's elapsed time from 3m36s to 10s), but doesn't eliminate the requirement to enable GitHub Actions. So I wouldn't recommend bothering with `.nojekyll` in general.
